datalad/distributed/tests/test_ora_http.py

```diff
@@ -0,0 +1,134 @@
+import shutil
+from datalad.api import (
+    Dataset,
+)
+from datalad.utils import Path
+from datalad.tests.utils import (
+    assert_equal,
+    assert_in,
+    assert_not_in,
+    assert_raises,
+    assert_repo_status,
+    assert_result_count,
+    known_failure_windows,
+    serve_path_via_http,
+    SkipTest,
+    with_tempfile
+)
+from datalad.distributed.ora_remote import (
+    LocalIO,
+)
+from datalad.support.exceptions import (
+    CommandError,
+)
+from datalad.distributed.tests.ria_utils import (
+    populate_dataset,
+    common_init_opts
+)
+from datalad.customremotes.ria_utils import (
+    create_store,
+    create_ds_in_store,
+)
+
+
+# NOTE: All we want and can test here is the HTTP functionality of the ORA
+#       remote. As of now, this is get and checkpresent only, sending one
+#       request each. The used URI for those requests is based on store layout
+#       version 1 and dataset layout version 2. Serving archives and/or
+#       different layouts via those requests is up to the server side, which we
+#       don't test here.
+
+@known_failure_windows  # see gh-4469
+@with_tempfile(mkdir=True)
+@serve_path_via_http
+@with_tempfile
+def test_initremote(store_path, store_url, ds_path):
+    ds = Dataset(ds_path).create()
+    store_path = Path(store_path)
+    url = "ria+" + store_url
+    init_opts = common_init_opts + ['url={}'.format(url)]
+
+    # fails on non-RIA URL
+    assert_raises(CommandError, ds.repo.init_remote, 'ora-remote',
+                  options=common_init_opts + ['url={}'
+                                              ''.format(store_path.as_uri())]
+                  )
+    # Doesn't actually create a remote if it fails
+    assert_not_in('ora-remote',
+                  [cfg['name']
+                   for uuid, cfg in ds.repo.get_special_remotes().items()]
+                  )
+
+    ds.repo.init_remote('ora-remote', options=init_opts)
+    assert_in('ora-remote',
+              [cfg['name']
+               for uuid, cfg in ds.repo.get_special_remotes().items()]
+              )
+    assert_repo_status(ds.path)
+    # git-annex:remote.log should have:
+    #   - url
+    #   - common_init_opts
+    #   - archive_id (which equals ds id)
+    remote_log = ds.repo.call_git(['cat-file', 'blob', 'git-annex:remote.log'])
+    assert_in("url={}".format(url), remote_log)
+    [assert_in(c, remote_log) for c in common_init_opts]
```

Comment:
Not a proper review, but I read through and didn't spot any obvious issues.

This line, though... what is this world coming to? :)

Reply:
With this line the world is coming to ... comprehension?

Reply:
Unlikely, that this would go wrong for just a single config option while not actually failing during initremote. Which is why I don't believe it makes a difference. But if I can make you happy by a proper loop, I will! :)

Reply:
yeah, I guess "comprehension" must be among valid answers to that question :)

[ Hitting the resolve button just to make it clear that I wasn't requesting a change here. ]